File: src/main/java/io/scriptor/type/StreamResultConverter.java
package io.scriptor.type;

import io.scriptor.http.result.HTTPResult;
import io.scriptor.http.result.HTTPResultStream;
import io.scriptor.http.result.HTTPResultVoid;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.InputStream;

public class StreamResultConverter implements IConverter<InputStream, HTTPResult<?>> {

    @Override
    public @NotNull HTTPResult<?> from(final @Nullable InputStream source) {
        if (source == null)
            return new HTTPResultVoid(200, "OK");
        return new HTTPResultStream(200, "OK", source);
    }
}

File: src/main/java/io/scriptor/type/StreamResultConverter.kt
package io.scriptor.type

import io.scriptor.http.result.HTTPResult
import io.scriptor.http.result.HTTPResultStream
import io.scriptor.http.result.HTTPResultVoid
import java.io.InputStream

class StreamResultConverter : IConverter<InputStream, HTTPResult<*>> {

    override fun from(source: InputStream?): HTTPResult<*> {
        if (source == null) {
            return HTTPResultVoid(200, "OK")
        }
        return HTTPResultStream(200, "OK", source)
    }
}
